lockfile/src/javascript.rs

```diff
@@ -67,19 +68,15 @@ impl Parse for PackageLock {
                     None => continue,
                 };
 
-                // Ignore bundled dependencies.
-                if keys.get("inBundle").is_some() {
-                    continue;
-                }
-
-                // Ignore extraneous dependencies.
-                if keys.get("extraneous").is_some() {
-                    continue;
-                }
-
                 // Get dependency type.
-                let resolved = get_field(keys, "resolved")
-                    .ok_or_else(|| anyhow!("Dependency '{name}' is missing \"resolved\" key"))?;
+                let resolved = match get_field(keys, "resolved") {
+                    Some(resolved) => resolved,
+                    // Ignore packages without clear resolution details.
+                    None => {
+                        debug!("ignoring package without `resolved` field: {name}");
```

Comment:
While testing this branch against the lockfile that generated this PR, I could not get this debug line to show in the output. Is there a specific configuration/environment/etc. required to see the debug statement?

```
❯ target/debug/phylum --version
phylum v6.6.4-3-gcac3dbf

❯ target/debug/phylum -vvv parse package-lock.json > parse_output.json
[2024-07-08T15:13:22Z DEBUG phylum_cli::config] Reading config from /Users/maxrake/.config/phylum/settings.yaml
[2024-07-08T15:13:22Z DEBUG phylum] Skipping update check...
[2024-07-08T15:13:22Z INFO  phylum_project] Found project configuration file at "/Users/maxrake/dev/phylum/localdev/cli/.phylum_project"
```

Reply:
`--skip-sandbox`